### PR TITLE
Register dependency serializers

### LINKED ISSUE #199: Webpack 5 caching not working (No serializer registered for IncludeDependency/AureliaDependency)
**I'm submitting a bug report**

* **Library Version:**
5.0.3

**Please tell us about your environment:**
* **Operating System:**
Debian

* **Node Version:**
v14.18.2

* **NPM Version:**
6.14.15

* **JSPM OR Webpack AND Version**
webpack: 5.65.0
webpack-cli: 4.9.1
webpack-dev-server 4.6.0

* **Browser:**
n/a

* **Language:**
TypeScript 4.5.3

**Current behavior:**

Running a build with the following webpack cache configuration:

```js
        cache: {
            type: "filesystem",
            buildDependencies: {
                config: [__filename]
            }
        }
```

I get many warnings like:

```
<w> [webpack.cache.PackFileCacheStrategy] Skipped not serializable cache item 'Compilation/modules|/home/glen/development-environment/Projects/Crew Panel/node_modules/ts-loader/index.js??ruleSet[1].rules[6].use[0]!/home/glen/development-environment/Projects/Crew Panel/src/app/resources/elements/gg-sort-link.ts': No serializer registered for IncludeDependency
<w> while serializing webpack/lib/cache/PackFileCacheStrategy.PackContentItems -> webpack/lib/NormalModule -> Array { 16 items } -> IncludeDependency
<w> [webpack.cache.PackFileCacheStrategy] Skipped not serializable cache item 'Compilation/modules|/home/glen/development-environment/Projects/Crew Panel/node_modules/ts-loader/index.js??ruleSet[1].rules[6].use[0]!/home/glen/development-environment/Projects/Crew Panel/src/app/view-models/app.ts': No serializer registered for AureliaDependency
<w> while serializing webpack/lib/cache/PackFileCacheStrategy.PackContentItems -> webpack/lib/NormalModule -> Array { 182 items } -> AureliaDependency
```

**Expected/desired behavior:**

* **What is the expected behavior?**

Support for webpack 5 caching / no warnings.

* **What is the motivation / use case for changing the behavior?**

Performance / clean output.

### PR DESCRIPTION
**Pull request by elitastic**
Fixes #199, I added dependency serializers, this adds support for filesystem cache in webpack 5.

### CODE REVIEW

File: src/AureliaDependenciesPlugin.ts [hunk 1]
```diff
@@ -21,8 +22,22 @@ class AureliaDependency extends IncludeDependency {
   get [dependencyImports]() {
     return webpack.Dependency.EXPORTS_OBJECT_REFERENCED as any;
   }
+
+  serialize(context: any) {
```
Comment (bigopon) [inline src/AureliaDependenciesPlugin.ts:26]: do we know the type of this context? if we do, I'd prefer that we have it here instead of any

**Review by bigopon — COMMENTED** [review]
Nice one, thanks. Can we have someway to test the serialization behavior? Anything basic will do

File: src/AureliaDependenciesPlugin.ts (hunk 1 above)
Comment (elitastic) [reply src/AureliaDependenciesPlugin.ts:26]: It's `ObjectSerializerContext `. Unfortunately, this type is only listed but not exported in the [webpack ts definition file](https://github.com/webpack/webpack/blob/main/types.d.ts) if I see this correctly.

**Comment by elitastic** [comment]
I added a very simple test with filesystem cache enabled. Tests in pipeline are broken, is this caused by me?

"webpack" command seems to be missing in command line?

> webpack --stats-error-details "--no-stats"
> 'webpack' is not recognized as an internal or external command,

```
npm install
npm WARN aurelia-webpack-plugin@5.0.3 requires a peer of webpack@>= 5.0.0 but none is installed. You must install peer dependencies yourself.
npm WARN aurelia-chart@0.3.4 requires a peer of chart.js@^3.1.0 but none is installed. You must install peer dependencies yourself.
npm WARN app-basic@ No description
npm WARN app-basic@ No repository field.
npm WARN app-basic@ No license field.
added 1 package from 4 contributors and audited 17 packages in 2.415s
71 packages are looking for funding
  run `npm fund` for details
found 0 vulnerabilities
> app-basic@ build:ci D:\a\webpack-plugin\webpack-plugin\tests\app-basic
> npm run build -- --no-stats
> app-basic@ build D:\a\webpack-plugin\webpack-plugin\tests\app-basic
> webpack --stats-error-details "--no-stats"
'webpack' is not recognized as an internal or external command,
operable program or batch file.
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! app-basic@ build: `webpack --stats-error-details "--no-stats"`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the app-basic@ build script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.
npm ERR! A complete log of this run can be found in:
npm ERR!     C:\npm\cache\_logs\2022-05-05T07_44_3[9](https://github.com/aurelia/webpack-plugin/runs/6302429069?check_suite_focus=true#step:5:9)_065Z-debug.log
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! app-basic@ build:ci: `npm run build -- --no-stats`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the app-basic@ build:ci script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.
npm ERR! A complete log of this run can be found in:
npm ERR!     C:\npm\cache\_logs\2022-05-05T07_44_39_[10](https://github.com/aurelia/webpack-plugin/runs/6302429069?check_suite_focus=true#step:5:10)1Z-debug.log
Error: Process completed with exit code 1.
```

**Comment by elitastic** [comment]
1. create empy aurelia 1 project `au new`
2. paste / override app.ts/app.html/my-component.ts/my-component.html
3. enable cache in webpack.config.js
```
cache: {
    type: "filesystem",
    buildDependencies: {
      config: [__filename]
    }
  },
```
4. npm start -> "my-component" is found and rendered
5. wait 20-30 seconds until webpack has finishes building the cache -> should have files in `node_modules/.cache/webpack`
6. npm start -> "my-component" is not rendered and cannot be found, see error in console

**Comment by bigopon** [comment]
@elitastic thanks, I'm able to see the errors, though I'm not sure what could be the cause. Totally unfamiliar with this class of issues. Are you able to help find a fix for it?

**Comment by elitastic** [comment]
Issue is fixed. `preserveModuleName` was not serialized on `NormalModule`.

**Pull request merged**